src/ui/pages/AssetInfo/AssetHeader.tsx

```diff
@@ -1,42 +1,192 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
 import { useCurrency } from 'src/modules/currency/useCurrency';
 import type { Asset } from 'src/modules/zerion-api/requests/asset-get-fungible-full-info';
 import { formatPriceValue } from 'src/shared/units/formatPriceValue';
-import { emDash } from 'src/ui/shared/typography';
+import VerifiedIcon from 'jsx:src/ui/assets/verified.svg';
 import { HStack } from 'src/ui/ui-kit/HStack';
 import { TokenIcon } from 'src/ui/ui-kit/TokenIcon';
 import { UIText } from 'src/ui/ui-kit/UIText';
+import { VStack } from 'src/ui/ui-kit/VStack';
+import { emitter } from 'src/ui/shared/events';
+import { emDash } from 'src/ui/shared/typography';
+import { useWindowSizeStore } from 'src/ui/shared/useWindowSizeStore';
+import * as styles from './styles.module.css';
 
-export function AssetHeader({
-  asset,
-  className,
-}: {
-  asset: Asset;
-  className?: string;
-}) {
+export function AssetHeader({ asset }: { asset: Asset }) {
   const { currency } = useCurrency();
+  const { innerWidth } = useWindowSizeStore();
+  const isUntrackedAsset = asset.meta.price == null;
+  const priceElementRef = useRef<HTMLDivElement>(null);
+  const symbolPlaceholderElementRef = useRef<HTMLDivElement>(null);
+  const dashPlaceholderElementRef = useRef<HTMLDivElement>(null);
+  const pricePlaceholderElementRef = useRef<HTMLDivElement>(null);
+
+  const [iconOffset, setIconOffset] = useState(0);
+  const [symbolOffset, setSymbolOffset] = useState(0);
+  const [dashOffset, setDashOffset] = useState(0);
+  const [priceOffset, setPriceOffset] = useState(0);
```

Comment:
```suggestion
const [offsets, setOffsets] = useState({
  iconOffset: 0,
  symbolOffset: 0,
  dashOffset: 0,
  priceOffset: 0,
});
```

Reply:
Maybe it makes sense to combine them into a single state object this this ^ wdyt?